app/controllers/candidate_interface/personal_details_controller.rb

```diff
@@ -21,14 +22,33 @@ def update
     end
 
     def show
-      personal_details_form = PersonalDetailsForm.build_from_application(
-        current_application,
-      )
+      personal_details_form = PersonalDetailsForm.build_from_application(@application_form)
       @personal_details_review = PersonalDetailsReviewPresenter.new(form: personal_details_form)
     end
 
+    def complete
+      if PersonalDetailsForm.build_from_application(@application_form).valid?
+        @application_form.update!(application_form_params)
+
+        redirect_to candidate_interface_application_form_path
+      else
+        flash[:warning] = 'You can’t mark this section as complete without adding all your personal details.'
+
+        @application_form.personal_details_completed = false
+
+        personal_details_form = PersonalDetailsForm.build_from_application(@application_form)
+        @personal_details_review = PersonalDetailsReviewPresenter.new(form: personal_details_form)
+
+        render :show
+      end
+    end
+
   private
 
+    def set_application_form_to_current_application
+      @application_form = current_application
+    end
```

Comment:
Can we use `current_application` in the code directly?

Reply:
We absolutely can. It's just used in pretty much all the views so I thought it would make sense to create it before. I'll continue to use current_applicatiion in the controller actions.